MapboxCoreNavigation/OfflineService.swift

```diff
@@ -0,0 +1,379 @@
+        }
+    }
+
+    private func notifyAvailable(forObserver observer: OfflineServiceObserver) {
+        DispatchQueue.main.async { [weak self] in
+            self?.regions.forEach {
+                if $0.mapsPack != nil {
+                    observer.didBecomeAvailable(region: $0, forDomain: .maps)
+                }
+                if $0.navigationPack != nil {
+                    observer.didBecomeAvailable(region: $0, forDomain: .navigation)
+                }
+            }
+        }
+    }
+}
+
+extension OfflineService: MapboxCommon.OfflineServiceObserver {
+    public func onPending(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack) {
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else { return }
+            let region = self._regions[metadata.id] ?? OfflineRegion(region: metadata)
+            self.observers.forEach {
+                $0.didAddPending(region: region, forDomain: .common(domain: domain))
+            }
+        }
+    }
+
+    public func onDownloading(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack) {
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else { return }
+            if let region = self._regions[metadata.id] {
+                if domain == .maps {
+                    region.mapsPack = OfflineRegionPack(pack: pack)
+                    region.mapsPack?.commonPackMetadata = metadata.mapPack
+                } else {
+                    region.navigationPack = OfflineRegionPack(pack: pack)
+                    region.navigationPack?.commonPackMetadata = metadata.navigationPack
+                }
+                self.observers.forEach {
+                    $0.didStartDownloading(region: region, forDomain: .common(domain: domain))
+                }
+            }
+        }
+    }
+
+    public func onIncomplete(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack) {
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else { return }
+            let region = self._regions[metadata.id] ?? OfflineRegion(region: metadata)
+            self.observers.forEach {
+                $0.didBecomeIncomplete(region: region, forDomain: .common(domain: domain))
+            }
+        }
+    }
+
+    public func onVerifying(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack) {
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else { return }
+            let region = self._regions[metadata.id] ?? OfflineRegion(region: metadata)
+            self.observers.forEach {
+                $0.didBeginVerifying(region: region, forDomain: .common(domain: domain))
+            }
+        }
+    }
+
+    public func onAvailable(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack) {
```

Comment:
In case of offline mode already downloaded regions are not shown, can it be addressed to match previous implementation?

Reply:
can't reproduce, could you provide more info about the case? It do show downloaded regions without an internet connection